x-pack/plugins/security_solution/public/detections/components/callouts/callout_types.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export type CallOutType = 'primary' | 'success' | 'warning' | 'danger';
+
+export interface CallOutMessage {
+  type: CallOutType;
+  id: string;
+  title: string;
+  description: JSX.Element;
+}
```

Comment:
Let's say (if) I wanted to share these types between:

- some components and some in-browser logic, what would be a good place for it?
- some components and some server-side logic?
- both?

This question is related to "how do we structure components and business logic they use?"

Reply:
If the types are going to be shared between/used in both client-side and server-side code, they should go in `common/`. If they're more localized to front end or even a specific component, I usually just have the type file in the same folder as the primary code its used in. Others might have different opinions on that

Reply:
I should specify, the top level `common` outside of `public`. There's an linter rule preventing `server` code from importing `public` code so shared types have gotta be outside them both

Reply:
Thank you. Makes sense.

> I usually just have the type file in the same folder as the primary code its used in

Agree as well. In general, that keeps cohesion higher. Of course, as always, it depends. :)